src/Analyzers/CSharp/Tests/UseIsNullCheck/UseIsNullCheckOverIsObjectTests.cs

```diff
@@ -0,0 +1,160 @@
+using Microsoft.CodeAnalysis.CSharp.UseIsNullCheck;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.UseIsNullCheck
+{
+    using VerifyCS = CSharpCodeFixVerifier<CSharpUseIsNullCheckOverIsObject, Testing.EmptyCodeFixProvider>;
+
+    [Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+    public class UseIsNullCheckOverIsObjectTests
+    {
+        private static async Task VerifyAsync(string source, string fixedSource, LanguageVersion languageVersion)
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = source,
+                FixedCode = fixedSource,
+                LanguageVersion = languageVersion,
+            }.RunAsync();
+        }
+
+        private static async Task VerifyCSharp9Async(string source, string fixedSource)
+            => await VerifyAsync(source, fixedSource, LanguageVersion.CSharp9);
+
+        private static async Task VerifyCSharp8Async(string source, string fixedSource)
+            => await VerifyAsync(source, fixedSource, LanguageVersion.CSharp8);
+
+        [Fact]
+        public async Task TestIsObjectCSharp8()
+        {
+            var source = @"
+public class C
+{
+    public bool M(string value)
+    {
+        return value is object;
+    }
+}
+";
+            await VerifyCSharp8Async(source, source);
+        }
+
+        [Fact]
+        public async Task TestIsObject()
+        {
+            var source = @"
+public class C
+{
+    public bool M(string value)
+    {
+        return [|value is object|];
+    }
+}
+";
+            var fixedSource = @"
+public class C
+{
+    public bool M(string value)
+    {
+        return value is not null;
+    }
+}
+";
+            await VerifyCSharp9Async(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task TestIsNotObject()
+        {
+            var source = @"
+public class C
+{
+    public bool M(string value)
+    {
+        return value is [|not object|];
+    }
+}
+";
+            var fixedSource = @"
+public class C
+{
+    public bool M(string value)
+    {
+        return value is null;
+    }
+}
+";
+            await VerifyCSharp9Async(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task TestIsStringAgainstObject_NoDiagnostic()
+        {
+            var source = @"
+public class C
+{
+    public bool M(object value)
+    {
+        return value is string;
+    }
+}
+";
+            await VerifyCSharp9Async(source, source);
+        }
+
+        [Fact]
+        public async Task TestIsStringAgainstString()
+        {
+            // Currently no diagnostic, but a diagnostic is reasonable too.
+            var source = @"
+public class C
+{
+    public bool M(string value)
+    {
+        return value is string;
+    }
+}
+";
+            await VerifyCSharp9Async(source, source);
+        }
```

Comment:
@CyrusNajmabadi Let me know if you want to extend the rule beyond `is object`

Reply:
i would, yes.